### PR TITLE
Fix download shared photos

### PR DESCRIPTION
**Pull request by lodesmets**
Fix download shared photos: https://github.com/mib1185/py-synologydsm-api/issues/262

Shared photos need to be downloaded with FotoTeams api instead of Foto api

### CODE REVIEW

File: src/synology_dsm/api/photos/model.py [hunk 1]
```diff
@@ -21,3 +21,4 @@ class SynoPhotosItem:
     file_size: str
     thumbnail_cache_key: str
     thumbnail_size: str
+    owner_user_id: int
```
Comment (mib1185) [inline src/synology_dsm/api/photos/model.py:24]: i think we shoud add a bool like `is_shared` which is set to True when "owner_user_id == 0"

File: src/synology_dsm/api/photos/__init__.py [hunk 2]
```diff
@@ -127,7 +133,9 @@ async def download_item(self, item: SynoPhotosItem) -> bytes | None:
     async def download_item_thumbnail(self, item: SynoPhotosItem) -> bytes | None:
         """Download the given items thumbnail."""
         raw_data = await self._dsm.get(
-            self.THUMBNAIL_API_KEY,
+            self.THUMBNAIL_FOTOTEAM_API_KEY
+            if item.owner_user_id == 0
+            else self.THUMBNAIL_API_KEY,
```
Comment (mib1185) [inline src/synology_dsm/api/photos/__init__.py:138]: we should avoid such multiline attribute associations, just set a helper variable before, like:
```python
download_api = self.THUMBNAIL_API_KEY
if item.is_shared:
    download_api = self.THUMBNAIL_FOTOTEAM_API_KEY
```

File: src/synology_dsm/api/photos/__init__.py [hunk 3]
```diff
@@ -143,7 +151,9 @@ async def download_item_thumbnail(self, item: SynoPhotosItem) -> bytes | None:
     async def get_item_thumbnail_url(self, item: SynoPhotosItem) -> str:
         """Get the url of given items thumbnail."""
         return await self._dsm.generate_url(
-            self.THUMBNAIL_API_KEY,
+            self.THUMBNAIL_FOTOTEAM_API_KEY
+            if item.owner_user_id == 0
+            else self.THUMBNAIL_API_KEY,
```
Comment (mib1185) [inline src/synology_dsm/api/photos/__init__.py:156]: same as above

**Review by mib1185 — CHANGES_REQUESTED** [review]
Hi @lodesmets 
many thanks for bring this fix in 🤝 
I've left some comments and it would be great to have some tests for this

**Comment by lodesmets** [comment]
I also added a test

**Pull request merged**